plugins/chat/assets/javascripts/discourse/routes/chat-index.js

```diff
@@ -43,8 +48,8 @@ export default class ChatIndexRoute extends DiscourseRoute {
       }
     }
 
-    // We are on desktop. Check for a channel to enter and transition if so
-    const id = this.chat.getIdealFirstChannelId();
+    // We are on desktop. Check for last visited channel and transition if so
+    const id = this.currentUser.custom_fields.last_chat_channel_id;
```

Comment:
Happy to see this, feels much more predictable 👍

Reply:
yes same, this whole function was way too complex